### PR TITLE
[plugin.video.mmlive] 2019.3.28

### PR DESCRIPTION
**Pull request by eracknaphobia**
### Description

Fix for upstream changes
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.mmlive/addon.xml [hunk 1]
```diff
@@ -1,24 +1,27 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<addon id="plugin.video.mmlive" name="March Madness Live" version="2017.3.19" provider-name="eracknaphobia">
-  <requires>
-   <import addon="xbmc.python" version="2.25.0"/>      
-  </requires>
-  <extension point="xbmc.python.pluginsource" library="main.py">
+<addon id="plugin.video.mmlive" name="March Madness Live" version="2019.3.28" provider-name="eracknaphobia">
+    <requires>
+        <import addon="xbmc.python" version="2.25.0"/>
+        <import addon="script.module.simplejson" version="3.3.0"/>
```
Comment (Rechi) [inline plugin.video.mmlive/addon.xml]: This module isn't used by the add-on.

File: plugin.video.mmlive/addon.xml [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
+        <summary lang="en_GB">Lets you stream all the March Madness games live.</summary>
+        <description lang="en_GB">The National Collegiate Athletic Association (NCAA) Men's Division I Basketball
+            Tournament is a single-elimination tournament played each spring in the United States, currently featuring
+            68 college basketball teams, to determine the national championship of the major college basketball teams.
+        </description>
+        <disclaimer lang="en_GB">Requires a valid US cable / satellite subscription</disclaimer>
+        <language>en</language>
+        <platform>all</platform>
+        <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
+        <forum>http://forum.kodi.tv/showthread.php?tid=264888</forum>
```
Comment (Rechi) [inline plugin.video.mmlive/addon.xml]: `https://forum.kodi.tv/showthread.php?tid=264888`

File: plugin.video.mmlive/addon.xml [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+        <source>https://github.com/eracknaphobia/plugin.video.mmlive/tree/krypton</source>
```
Comment (Rechi) [inline plugin.video.mmlive/addon.xml]: Doesn't look like you took the code from krypton branch, `https://github.com/eracknaphobia/plugin.video.mmlive`

File: plugin.video.mmlive/main.py [hunk 4]
```diff
@@ -131,21 +131,33 @@ def setArchiveStreams(tourn_day, json_source, teams):
...
+        stream_url += '|User-Agent='+UA_MMOD
     else:
         adobe = ADOBE(SERVICE_VARS)        
         resource_id = 'truTV'
         mvpd = adobe.authorizeDevice(resource_id)        
         media_token = adobe.mediaToken(resource_id)          
-        playable_stream = tokenTurner(media_token,stream_url,mvpd)        
+        stream_url = tokenTurner(media_token,stream_url,mvpd)
+
+    xbmc.log('------------------------------------------------------------------------------------------')
```
Comment (Rechi) [inline plugin.video.mmlive/main.py]: log spam

File: plugin.video.mmlive/main.py [hunk 5]
```diff
@@ -131,21 +131,33 @@ def setArchiveStreams(tourn_day, json_source, teams):
...
+    xbmc.log(stream_url)
```
Comment (Rechi) [inline plugin.video.mmlive/main.py]: kodi logs that already

File: plugin.video.mmlive/main.py [hunk 6]
```diff
@@ -131,21 +131,33 @@ def setArchiveStreams(tourn_day, json_source, teams):
...
+    xbmc.log('------------------------------------------------------------------------------------------')
```
Comment (Rechi) [inline plugin.video.mmlive/main.py]: log spam

File: plugin.video.mmlive/resources/globals.py [hunk 7]
```diff
@@ -337,14 +314,13 @@ def getAuthCookie():
 def addStream(name,link_url,title,game_id,icon=None,fanart=None):
     ok=True
     u=sys.argv[0]+"?url="+urllib.quote_plus(link_url)+"&mode="+str(104)+"&name="+urllib.quote_plus(name)+"&game_id="+urllib.quote_plus(str(game_id))    
-        
-    liz=xbmcgui.ListItem(name)
-    liz.setArt({'icon': ICON, 'thumb': ICON})
+    
+    liz=xbmcgui.ListItem(name, iconImage=ICON, thumbnailImage=ICON)
```
Comment (Rechi) [inline plugin.video.mmlive/resources/globals.py]: `iconImage` and `thumbnailImage` are deprecated use `setArt(...)`.

File: plugin.video.mmlive/resources/globals.py [hunk 8]
```diff
@@ -353,36 +329,42 @@ def addStream(name,link_url,title,game_id,icon=None,fanart=None):
...
 
-def addDir(name,url,mode,icon,fanart=None):       
+def addDir(name,url,mode,iconimage,fanart=None):       
     ok=True    
     u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)
     
-    liz=xbmcgui.ListItem(name)
-    liz.setArt({'icon': icon, 'thumb': icon})
+    if iconimage != None:
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (Rechi) [inline plugin.video.mmlive/resources/globals.py]: `iconImage` and `thumbnailImage` are deprecated use `setArt(...)`.

File: plugin.video.mmlive/resources/globals.py [hunk 9]
```diff
@@ -353,36 +329,42 @@ def addStream(name,link_url,title,game_id,icon=None,fanart=None):
...
+def addDir(name,url,mode,iconimage,fanart=None):       
     ok=True    
     u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)
     
-    liz=xbmcgui.ListItem(name)
-    liz.setArt({'icon': icon, 'thumb': icon})
+    if iconimage != None:
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
+    else:
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON)
```
Comment (Rechi) [inline plugin.video.mmlive/resources/globals.py]: `iconImage` and `thumbnailImage` are deprecated use `setArt(...)`.

File: plugin.video.mmlive/resources/globals.py [hunk 10]
```diff
@@ -353,36 +329,42 @@ def addStream(name,link_url,title,game_id,icon=None,fanart=None):
...
 
 
-def addLink(name,url,icon,fanart=None):
+def addLink(name,url,iconimage,fanart=None):
     ok=True            
-    
-    liz=xbmcgui.ListItem(name)
-    liz.setArt({'icon': icon, 'thumb': icon})
+    if iconimage != None:
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (Rechi) [inline plugin.video.mmlive/resources/globals.py]: `iconImage` and `thumbnailImage` are deprecated use `setArt(...)`.

File: plugin.video.mmlive/resources/globals.py [hunk 11]
```diff
@@ -353,36 +329,42 @@ def addStream(name,link_url,title,game_id,icon=None,fanart=None):
...
-def addLink(name,url,icon,fanart=None):
+def addLink(name,url,iconimage,fanart=None):
     ok=True            
-    
-    liz=xbmcgui.ListItem(name)
-    liz.setArt({'icon': icon, 'thumb': icon})
+    if iconimage != None:
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
+    else:
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON)
```
Comment (Rechi) [inline plugin.video.mmlive/resources/globals.py]: `iconImage` and `thumbnailImage` are deprecated use `setArt(...)`.

File: plugin.video.mmlive/addon.xml [hunk 12]
```diff
@@ -1,24 +1,27 @@
...
-  <extension point="xbmc.addon.metadata">    
-    <platform>all</platform>
-    <summary lang="en_GB"></summary>
-    <description lang="en_GB">The National Collegiate Athletic Association (NCAA) Men's Division I Basketball Tournament is a single-elimination tournament played each spring in the United States, currently featuring 68 college basketball teams, to determine the national championship of the major college basketball teams.</description>
-    <disclaimer lang="en_GB"></disclaimer>
-    <language>en</language>
-    <platform>all</platform>
-    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
-    <forum>http://forum.kodi.tv/showthread.php?tid=264888</forum>
-    <website></website>
```
Comment (Rechi) [inline plugin.video.mmlive/addon.xml]: readd the website tag with the content provider site.

**Comment by Rechi** [comment]
Apply the following diff to use the print function syntax, even better would be `xbmc.log(...)`
```
--- .\plugin.video.mmlive\resources\globals.py  (original)
+++ .\plugin.video.mmlive\resources\globals.py  (refactored)
@@ -116,14 +116,14 @@
     if len(stream_title) > 0:
         ret =-1
         stream_title.sort(key=natural_sort_key)
-        print "PLAY BEST SETTING"
-        print PLAY_BEST
+        print("PLAY BEST SETTING")
+        print(PLAY_BEST)
         if str(PLAY_BEST) == 'true':
             ret = len(stream_title)-1
         else:
             dialog = xbmcgui.Dialog()
             ret = dialog.select('Choose Stream Quality', stream_title)
-            print ret
+            print(ret)
         if ret >=0:
             url = stream_url.get(stream_title[ret])
         else:
@@ -221,7 +221,7 @@

     for game in games:
         if str(game['id']) == str(game_id):
-            print game
+            print(game)
             clock = str(game['clock'])
             per = str(game['per'])
             state = str(game['state'])
```

Also explicitly declare the icon and fanart in the assets tag in addon.xml.

File: plugin.video.mmlive/addon.xml (hunk 2 above)
Comment (eracknaphobia) [reply plugin.video.mmlive/addon.xml]: Not sure I get why forum is incorrect? The link takes me to the correct site.

**Comment by Rechi** [comment]
@eracknaphobia this add-on isn't Python 3 compatible. Future updates have to be compatible with Python 3.
Have a look at https://kodi.tv/article/attention-addon-developers-migration-python-3 for more information.

**Pull request merged**